x-pack/plugins/ml/server/models/data_recognizer/data_recognizer.test.ts

```diff
@@ -4,13 +4,24 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { SavedObjectsClientContract, KibanaRequest } from 'kibana/server';
+import { SavedObjectsClientContract, KibanaRequest, IScopedClusterClient } from 'kibana/server';
 import { Module } from '../../../common/types/modules';
 import { DataRecognizer } from '../data_recognizer';
 
+const callAs = () => {
+  return new Promise((resolve) => {
+    resolve({ body: {} });
+  }) as Promise<any>;
+};
+
+const mlClusterClient = ({
+  asCurrentUser: callAs,
+  asInternalUser: callAs,
+} as unknown) as IScopedClusterClient;
```

Comment:
Can we create a shared mocked for the cluster client? Or maybe kibana provides one already

Reply:
yes, having a common mock for this would be very useful. in almost every test I had to change for this PR, the client had been mocked in a slightly different way.
Introducing that common mock now would have extended this work for longer than I wanted, but it would be great to do in the near future.